mars-sim-core/src/main/java/com/mars_sim/core/UnitManager.java

```diff
@@ -336,60 +314,26 @@ public Vehicle getVehicleByID(Integer id) {
 	 * @param unit new unit to add.
 	 */
 	public synchronized void addUnit(Unit unit) {
+		int unitIdentifier = unit.getIdentifier();
 
-		if (unit != null) {
-			switch(unit.getUnitType()) {
-			case SETTLEMENT:
-				lookupSettlement.put(unit.getIdentifier(),
-			   			(Settlement) unit);
-				addDisplayUnit(unit);
-				break;
-			case PERSON:
-				lookupPerson.put(unit.getIdentifier(),
-			   			(Person) unit);
-				break;
-			case ROBOT:
-				lookupRobot.put(unit.getIdentifier(),
-			   			(Robot) unit);
-				break;
-			case VEHICLE:
-				lookupVehicle.put(unit.getIdentifier(),
-			   			(Vehicle) unit);
-				addDisplayUnit(unit);
```

Comment:
We loose adding the Vehicles & Settlements to the displayUnit list but to be honest I hate that list. It artificial as it is just used in the Navigation Panel. The list is supporting functionality that is only used by Navigation panel so we should rework the latter.
So keep this part of the change. I've created #1408 to drop displayUnit list.

Reply:
@bevans2000 

Will let you decide what to change or keep.